### DISCUSSION TITLE
path reverse proxy using Nginx

### DISCUSSION THREAD

**Discussion by kexue-z** [requarks/wiki#6445] [Help / Questions]
I am using nginx as a reverse proxy for wiki.js on my website. I want to access it via https://mydomain.com:3001/wiki. Here is my nginx configuration:

```
server {
  listen 3001  ssl;
  listen [::]:3001 ssl;

  server_name mydomain.com;

  ssl_certificate  /ssl/mydomain.com.crt;
  ssl_certificate_key /ssl/mydomain.com.key;

  ssl_session_timeout 5m;
  ssl_ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE:ECDH:AES:HIGH:!NULL:!aNULL:!MD5:!ADH:!RC4;
  ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
  ssl_prefer_server_ciphers on;
  add_header Strict-Transport-Security "max-age=63072000; includeSubdomains; preload";
  client_max_body_size 10000m;

  location  /wiki {
    proxy_redirect off;
    proxy_pass http://127.0.0.1:3000;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Upgrade $http_upgrade; 
    proxy_set_header Connection "upgrade";
  }
}
```

I already have a website running on `/` and want to add wiki to it. Wiki.js runs well on `http://127.0.0.1:3000`, but shows errors on `https://mydomain.com:3001/wiki`.

Please let me know if you can help.

![image](https://github.com/requarks/wiki/assets/71873002/573b1c2a-c92d-43fc-b315-7bc208fb1ec6)

**Comment by fnafox** [comment]
It's not possible to point the wiki to a subdirectory, it has to be on webroot.

Seeing as you're referring to it by port, what is the use of rewriting it to the `/wiki` subdirectory? Why not just use `https://mydomain.com:3001/`?

Though in this case I don't really see the use for a reverse proxy since you could just set the port for your wiki to 3001 in the first place. If you are using a proxy, why not set up a subdomain `wiki.mydomain.com` and avoid the need for port specification entirely?

**Reply by kexue-z** [reply]
Since I am using my company's server, it only allows me to use a specific port to open to the public. And subdomains may not be easy to obtain. Therefore, consider using this method.

**Reply by fnafox** [reply]
Unfortunately there's no way to make wiki.js work in a subdirectory as far as I know.

**Reply by kexue-z** [reply]
Ok, I may find another way.
